plugins/dead_code_elimination/dead_code_elimination_main.ml

```diff
@@ -73,27 +75,72 @@ let propagate_consts sub =
       Map.fold vars ~init:exp ~f:(fun ~key:pat ~data:rep ->
           Exp.substitute Bil.(var pat) rep)))
 
-let rec process proj =
-  let prog = Project.program proj in
-  report_progress ~note:"ssa" ();
-  let subs = Term.enum sub_t prog |> Seq.map ~f:Sub.ssa in
+let is_alive dead t = not (Set.mem dead (Term.tid t))
+let live_def dead blk = Term.filter def_t ~f:(is_alive dead) blk
+let live_phi dead blk = Term.filter phi_t ~f:(is_alive dead) blk
+
+let clean dead ssa =
```

Comment:
the clean procedure was doing `propagate_consts` before, which actually affects the dead code elimination procedure,  for example:

```
x := 12
y := x + 1
```
after constant propagation became

```
x := 12
y := 12 + 1
```

That makes `x` dead, so it could be removed.